build.py

```diff
@@ -0,0 +1,418 @@
+        self.provisioning = provisioning
+        self.ansible_args = ansible_args
+        self.image_name = self.assemble_name()
+        self.image_path = pathlib.Path(DIR_PATH + "/" + self.image_name + ".raw")
+        if conda_env:
+            self.qemu_path = f"{conda_env}/bin/qemu-img"
+            self.openstack_path = f"{conda_env}/bin/openstack"
+            self.packer_path = f"{conda_env}/bin/packer"
+        else:
+            self.packer_path = shutil.which("packer")
+            self.openstack_path = shutil.which("openstack")
+            self.qemu_path = shutil.which("qemu-img")
+
+    def dry_run(self):
+        print(self.assemble_packer_envs())
+        print(self.assemble_packer_build_command())
+        print(self.image_name)
+        print(self.assemble_convert_command())
+        if self.openstack:
+            print(self.assemble_os_command())
+        if self.pvt_key:
+            print(self.assemble_scp_command())
+            print(self.assemble_ssh_command())
+
+    def assemble_packer_init(self):
+        return [
+            f"self.packer_path",
+            f"init",
+            f"{DIR_PATH}/templates",
+        ]
+
+    def assemble_packer_build_command(self):
+        return " ".join(
+            [
+                f"{self.packer_path}",
+                f"build",
+                f"-only=qemu.{self.template}",
+                f"{DIR_PATH}/templates",
+            ]
+        )
+
+    def assemble_convert_command(self):
+        return " ".join(
+            [
+                f"{self.qemu_path}",
+                f"convert",
+                f"-O",
+                f"raw",
+                f"./images/{self.template}",
+                f"{self.image_path}",
+            ]
+        )
+
+    def assemble_os_command(self):
+        return " ".join(
+            [
+                f"{self.openstack_path}",
+                f"image",
+                f"create",
+                f"--file",
+                f"{self.image_path}",
+                f"{self.image_name}",
+            ]
+        )
+
+    def assemble_packer_envs(self):
+        env = os.environ.copy()
+        env["PACKER_PLUGIN_PATH"] = f"{DIR_PATH}/packer_plugins"
+        env[
+            "PKR_VAR_groups"
+        ] = f"[{','.join('\"' + x + '\"' for x in self.provisioning)}]"
+        env["PKR_VAR_headless"] = "true"
+        if self.ansible_args:
+            env["PKR_VAR_ansible_extra_args"] = self.ansible_args
+        return env
+
+    def assemble_name(self):
+        """
+        Uses a naming scheme described in\n
+        https://github.com/usegalaxy-eu/vgcn/issues/78
+        """
+        name = ["vgcn"]
+        if "generic" in self.provisioning:
+            provisioning = self.provisioning
+            provisioning.remove("generic")
+            name += [f"+{'+'.join(provisioning)}"]
+        else:
+            name += [f"!generic+{'+'.join(self.provisioning)}"]
+        name += [
+            "-".join(self.template.split("-", 2)[:2]),
```

Comment:
Doesn't read as nice, but I am happy to consider the architecture as part of the OS. Let's say my initial intention was to have the Packer build name there rather than the OS, sorry for the misunderstanding.

```suggestion
            self.template,
```